### DISCUSSION TITLE
Logs to file

### DISCUSSION THREAD

**Discussion by infernalx** [Kludex/uvicorn#1418] [Q&A] [answered]
Is there a simple way to save logs from terminal into file ?

I have formatted adding timestamp on it but no idea how to save it  and is it bad running uvicorn in production i read on page that i should be using gunicorn but then im not sure what to run anymore.

**Comment by euri10** [comment, marked as answer]
you probably want to use a `FileHandler` (see https://docs.python.org/3/library/logging.handlers.html) in your logging config and use it with our `--log-config` flag (see https://www.uvicorn.org/settings/#logging)

**Reply by infernalx** [reply]
If i understand right i should put it somewhere in log config but i have no idea how it should look, here is the current log conf which i have 
[log_conf.txt](https://github.com/encode/uvicorn/files/8349545/log_conf.txt)

**Reply by euri10** [reply]
you need to 1) change your logging configuration extension to either json or yaml, as stated in the docs I just linked above, 2) use a FileHandler in your log_conf handlers section as stated in the link above, 3) rtfm

**Reply by infernalx** [reply]
i mean my file is yaml but cant upload yaml here and if i post it as code it would be in single line so not readable thats why i put it as txt but i dont know how to properly include filehandler thats why i replied again so if you can provide me with some more help that would be nice if not thanks anyway

**Reply by infernalx** [reply]
so i added into handlers
`  file_handler:
    class: logging.FileHandler
    level: INFO
    filename: python_service.log
    encoding: utf8
    mode: a`
file appears but its not logging anything its empty did i do something wrong or need to do something extra ?
